### PR TITLE
sshdriver: Log keepalive output

### PR DESCRIPTION
**Pull request by JoshuaWatt**
While it is not normally expected for the keepalive process to produce any output, it can be helpful for debugging purposes to log any output it may have produced. As such, try harder to get output when terminating the process, and also report any output after it exits

**Description**

**Checklist**
- [ ] Documentation for the feature
- [ ] Tests for the feature 

- [ ] The arguments and description in doc/configuration.rst have been updated

- [ ] Add a section on how to use the feature to doc/usage.rst

- [ ] Add a section on how to use the feature to doc/development.rst

- [x] PR has been tested

- [ ] Man pages have been regenerated

### CODE REVIEW

**Review by Emantor — DISMISSED** [review]

File: labgrid/driver/sshdriver.py [hunk 1]
```diff
@@ -507,16 +507,35 @@ def _check_keepalive(self):
...
-        try:
-            self._keepalive.wait(timeout=60)
+            try:
+                # Try again to get output
+                stdout, stderr = self._keepalive.communicate(timeout=60)
+            except subprocess.TimeoutExpired:
+                try:
+                    self._keepalive.wait(timeout=60)
+                except subprocess.TimeoutExpired:
+                    self.logger.warning("ssh keepalive for %s timed out during termination", self.networkservice.address)
```
Comment (Bastian-Krause) [inline labgrid/driver/sshdriver.py:536]: Why is that needed? Using `communicate` instead of `wait` should not change anything, so adding this is surprising to me.

File: labgrid/driver/sshdriver.py [hunk 2]
```diff
@@ -507,16 +507,35 @@ def _check_keepalive(self):
...
         finally:
             self._keepalive = None
+
+        log_output("stdout", stdout)
+        log_output("stderr", stderr)
```
Comment (Bastian-Krause) [inline labgrid/driver/sshdriver.py:541]: As you said, normally no output is generated. Is it worth to distinguish stdout/stderr here? Maybe change the `subprocess.Popen` in `_start_keepalive()` (stdin isn't used):

```diff
         self._keepalive = subprocess.Popen(
             args,
-            stdin=subprocess.PIPE,
             stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
         )
 ```
If you also specify `encoding="utf-8"` there, you can probably also drop `log_output()` and do the logging here? The logging could even happen in the `try..except`, so you don't have to initialize the variables beforehand. This would slim down the patch even more.

**Review by Bastian-Krause — CHANGES_REQUESTED** [review]
Seems useful, see some suggestions to slim down the patch below.

File: labgrid/driver/sshdriver.py (hunk 2 above)
Comment (JoshuaWatt) [reply labgrid/driver/sshdriver.py]: stdin has to be `subprocess.PIPE` because this is how the keep alive is terminated (closing stdin), but otherwise I think I've done all of these.

**Review by Bastian-Krause — APPROVED** [review]
Thanks!

**Review by Emantor — APPROVED** [review]

**Comment by Emantor** [comment]
Thanks!
